tests/MongoDB.Analyzer.Tests/Core/PocoDataFillerTests.cs

```diff
@@ -0,0 +1,121 @@
+    }
+
+    [DataTestMethod]
+    public void TestCollectionHolder()
+    {
+        var poco = new ListsHolder();
+        PocoDataFiller.PopulatePoco(poco);
+
+        CollectionAssert.AreEqual(poco.PesonsList, new List<Person>(0));
+        CollectionAssert.AreEqual(poco.StringList, new List<string>(0));
+        CollectionAssert.AreEqual(poco.NestedListsHolderList, new List<ListsHolder>(0));
+        CollectionAssert.AreEqual((List<int>)poco.IntIList, new List<int>(0));
+        CollectionAssert.AreEqual((List<int>)poco.IntIList, new List<int>(0));
+        CollectionAssert.AreEqual((List<ListsHolder>)poco.NestedListsHolderIList, new List<ListsHolder>(0));
+    }
+
+    [DataTestMethod]
+    public void TestSimpleArrayHolder()
+    {
+        var poco = new SimpleTypesArraysHolder();
+        PocoDataFiller.PopulatePoco(poco);
+        CollectionAssert.AreEqual(poco.ByteArray, new byte[] {});
+        CollectionAssert.AreEqual(poco.IntArray, new int[] { });
+        CollectionAssert.AreEqual(poco.ObjectArray, new object[] { });
+        CollectionAssert.AreEqual(poco.JaggedStringArray2, new string[][] { });
+        CollectionAssert.AreEqual(poco.JaggedIntArray3, new int[][][] { });
+        CollectionAssert.AreEqual(poco.JaggedLongArray4, new long[][][][] { });
+        CollectionAssert.AreEqual(poco.JaggedShortArray5, new short[][][][][] { });
+    }
+
+    [DataTestMethod]
+    public void TestMultiDimentionalArrayHolder()
+    {
+        var poco = new MultiDimentionalArrayHolder();
+        PocoDataFiller.PopulatePoco(poco);
+        CollectionAssert.AreEqual(poco.Matrix2, new int[,] { });
+        CollectionAssert.AreEqual(poco.Matrix3, new int[,,] { });
+    }
+
+    [DataTestMethod]
+    public void TestPrimitiveTypeHolder()
+    {
+        var poco = new PrimitiveTypeHolder();
+        PocoDataFiller.PopulatePoco(poco);
+        Assert.AreEqual(poco.BooleanValue, true);
+        Assert.AreEqual(poco.ByteValue, 9);
+        Assert.AreEqual(poco.SByteValue, 0);
+        Assert.AreEqual(poco.ShortValue, 0);
+        Assert.AreEqual(poco.UShortValue, 1);
+        Assert.AreEqual(poco.IntValue, 8);
+        Assert.AreEqual(poco.UIntValue, (uint)9);
+        Assert.AreEqual(poco.LongValue, 9);
+        Assert.AreEqual(poco.ULongValue, (ulong)0);
+        Assert.AreEqual(poco.CharValue, 9);
+        Assert.AreEqual(poco.DoubleValue, 1.0);
+        Assert.AreEqual(poco.StringValue, "StringValue");
+        Assert.AreEqual(poco.FloatValue, 0.0);
+    }
+
+    [DataTestMethod]
+    public void TestPropertiesAndFields()
+    {
+        var poco = new Fruit();
+        PocoDataFiller.PopulatePoco(poco);
+        Assert.AreEqual(poco.Name, "Name");
+        Assert.AreEqual(poco.Weight, 6.0);
+        Assert.AreEqual(poco.Color, "Color");
+        Assert.AreEqual(poco.Quantity, 8);
+        Assert.AreEqual(poco.TotalCost, 9.0);
+        Assert.AreEqual(poco.Volume, 6.0);
+    }
```

Comment:
Test also 
1. MaxLlevel 
2. Self referencing pocos.

Reply:
Included a test in PocoDataFillerTests that essentially covers both of them in one test method. The POCO is self-referencing and I test to make sure that this obj is populated only until a certain level.